draft-pda-protocol.md

```diff
@@ -1001,31 +1007,70 @@ indefinitely until a suitable output is found.
 
 [TODO: Define Hits-specific protocol messages.]
 
-# System design
+# Operational Considerations {#operational-capabilities}
+
+PDA protocols have inherent constraints derived from the tradeoff between privacy
+guarantees and computational complexity. These tradeoffs influence how
+applications may choose to utilize services implementing the specification.
+
+## Protocol participant capabilities {#entity-capabilities}
 
-[[OPEN ISSUE: This section seems like a catch-all for things not in other
-sections. Perhaps there is a natural home for aggregator discovery, share
-uploading, open issues, and system parameters?]]
+The design in this document has different assumptions and requirements for
+different protocol participants, including clients, aggregators, and
+collectors. This section describes these capabilities in more detail.
 
-## Aggregator discovery
+### Client capabilities
 
-[[OPEN ISSUE: writeme]]
+Clients have limited capabilities and requirements. Their only inputs to the protocol
+are (1) the PDAParam structure configured out of band and (2) a measurement. Clients
+are not expected to store any state across any upload
+flows, nor are they required to implement any sort of report upload retry mechanism.
+By design, the protocol in this document is robust against individual client upload
+failures since the protocol output is an aggregate over all inputs.
 
-## Share uploading
+### Aggregator capabilities
 
-[[OPEN ISSUE: writeme]]
+Helpers and leaders have different operational requirements. The design in this
+document assumes an operationally competent leader, i.e., one that has no storage
+or computation limitations or constraints, but only a modestly provisioned helper, i.e., one that
+has computation, bandwidth, and storage constraints. By design, leaders must be
+at least as capable as helpers, where helpers are generally required to:
 
-## Open questions and system parameters {#questions-and-params}
+- Support the collect protocol, which includes verifying and aggregating
+  sets of reports in a given batch; and
+- Publish and manage an HPKE configuration that can be used for the upload protocol;
 
-[[OPEN ISSUE: discuss batch size parameter and thresholds]]
-[[OPEN ISSUE: discuss f^ leakage differences from [GB17]]]
+In addition, for each PDAParam instance, helpers are required to:
 
+- Implement some form of batch-to-report index, as well as inter- and intra-batch
+  replay mitigation storage, which includes some way of tracking batch report size
+  with optional support for state offloading. Some of this state may be used for
+  replay attack mitigation. One relay mitigation strategy is described in {{anti-replay}}.
```

Comment:
```suggestion
  replay attack mitigation. One replay mitigation strategy is described in {{anti-replay}}.
```
Also, isn't the strategy in {{anti-replay}} *the* strategy? Do we intend to allow deployments to skip this, or do something else?